discovery/src/lib.rs

```diff
@@ -129,12 +138,22 @@ impl Builder {
         )?;
 
         #[cfg(not(feature = "with-dns-sd"))]
-        let svc = libmdns::Responder::spawn(&tokio::runtime::Handle::current())?.register(
-            "_spotify-connect._tcp".to_owned(),
-            name,
-            port,
-            &["VERSION=1.0", "CPath=/"],
-        );
+        let svc = if !_bind_ips.is_empty() {
```

Comment:
Since it's [`libmdns::Responder`](https://docs.rs/libmdns/latest/libmdns/struct.Responder.html) uses a builder pattern, you could DRY up this code to 

```rust
        let svc = if !_bind_ips.is_empty() {
            libmdns::Responder::spawn_with_ip_list(&tokio::runtime::Handle::current(), _bind_ips)                
        } else {
            libmdns::Responder::spawn(&tokio::runtime::Handle::current())
        };        

        Ok(Discovery { server, _svc: svc?.register(
                    "_spotify-connect._tcp".to_owned(),
                    name,
                    port,
                    &["VERSION=1.0", "CPath=/"],
                ) })
    }
```

Reply:
Thanks for the hint, I will change it.